charts/internal/alicloud-infra/templates/main.tf

```
@@ -45,45 +45,39 @@ resource "alicloud_vswitch" "vsw_z{{ $index }}" {
   }
 }
 
-{{- $createEip := true }}
-{{ if $zone.natGateway -}}
-{{ if $zone.natGateway.eipAllocationID }}
-// specify EIP id
+{{ if $zone.eipAllocationID -}}
+// specify EIP ID
 data "alicloud_eips" "eip_natgw_ds{{ $index }}" {
-  ids = ["{{ $zone.natGateway.eipAllocationID }}"]
+  ids = ["{{ required "eipAllocationID is required" $zone.eipAllocationID }}"]
 }
 
 resource "alicloud_eip_association" "eip_natgw_asso_z{{ $index }}" {
   allocation_id = data.alicloud_eips.eip_natgw_ds{{ $index }}.eips.0.id
-  instance_id   = {{ required "natGatewayID is required" $.Values.vpc.natGatewayID }}
+  instance_id   = {{ required "natGatewayID is required" $.Values.natGateway.natGatewayID }}
 }
 
 resource "alicloud_snat_entry" "snat_z{{ $index }}" {
-  snat_table_id     = {{ required "snatTableID is required" $.Values.vpc.snatTableID }}
+  snat_table_id     = {{ required "sNatTableIDs is required" $.Values.natGateway.sNatTableIDs }}
   source_vswitch_id = alicloud_vswitch.vsw_z{{ $index }}.id
   snat_ip           = data.alicloud_eips.eip_natgw_ds{{ $index }}.eips.0.ip_address
   depends_on        = [alicloud_eip_association.eip_natgw_asso_z{{ $index }}]
 }
-{{- $createEip = false }}
-{{- end }}
-{{- end }}
-
-{{- if $createEip }}
+{{ else -}}
 // Create a new EIP.
 resource "alicloud_eip" "eip_natgw_z{{ $index }}" {
   name                 = "{{ required "clusterName is required" $.Values.clusterName }}-eip-natgw-z{{ $index }}"
   bandwidth            = "100"
   instance_charge_type = "PostPaid"
-  internet_charge_type = "{{ required "vpc.internetChargeType is required" $.Values.vpc.internetChargeType }}"
+  internet_charge_type = "{{ required "eip.internetChargeType is required" $.Values.eip.internetChargeType }}"
 }
 
 resource "alicloud_eip_association" "eip_natgw_asso_z{{ $index }}" {
   allocation_id = alicloud_eip.eip_natgw_z{{ $index }}.id
-  instance_id   = {{ required "natGatewayID is required" $.Values.vpc.natGatewayID }}
+  instance_id   = {{ required "natGatewayID is required" $.Values.natGateway.natGatewayID }}
 }
 
 resource "alicloud_snat_entry" "snat_z{{ $index }}" {
-  snat_table_id     = {{ required "snatTableID is required" $.Values.vpc.snatTableID }}
+  snat_table_id     = {{ required "sNatTableIDs is required" $.Values.natGateway.sNatTableIDs }}
```

```suggestion
  snat_table_id     = {{ required "natGateway.sNatTableIDs is required" $.Values.natGateway.sNatTableIDs }}
```

(same for other `required` statements above)